src/terms/xc.jl

```diff
@@ -309,45 +223,102 @@ function LibxcDensity(basis, max_derivative::Integer, ρ::RealFourierArray, ρsp
             end
         end
 
+        # Does the spin index transformation (σ, τ) => στ as expected by Libxc
+        tσ = libxc_spinindex_σ
+
         σ_real .= 0
         @views for α in 1:3
-            # Does the index transformation (σ, τ) => στ like in libxc_symmetric_index
-            σ_real[1, :, :, :] .+= ∇ρ_real[1, :, :, :, α] .* ∇ρ_real[1, :, :, :, α]
+            σ_real[tσ(1, 1), :, :, :] .+= ∇ρ_real[1, :, :, :, α] .* ∇ρ_real[1, :, :, :, α]
             if n_spin > 1
-                σ_real[2, :, :, :] .+= ∇ρ_real[1, :, :, :, α] .* ∇ρ_real[2, :, :, :, α]
-                σ_real[3, :, :, :] .+= ∇ρ_real[2, :, :, :, α] .* ∇ρ_real[2, :, :, :, α]
+                σ_real[tσ(1, 2), :, :, :] .+= ∇ρ_real[1, :, :, :, α] .* ∇ρ_real[2, :, :, :, α]
+                σ_real[tσ(2, 2), :, :, :] .+= ∇ρ_real[2, :, :, :, α] .* ∇ρ_real[2, :, :, :, α]
             end
```

Comment:
Or maybe we should add one extra layer of indirection on top of the returned `terms` object, which does the mapping and adds the strange (1/2) factor automatically, only returning a view into the 3D real-space grid. That could help to make the contractions more readable (because it's just elementwise multiplications and sums so no further indexing needed), but it hides which indices are contiguous.

Reply:
I'll leave that when merging the branches, but I think this might be the way to go.